pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -162,14 +162,15 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(builder *ListenerBui
 		listeners = append(listeners, mutable.Listener)
 	}
 	// We'll try to return any listeners we successfully marshaled; if we have none, we'll emit the error we built up
-	err := errs.ErrorOrNil()
-	if err != nil {
+	if len(errs.Errors) == 1 {
 		// we have some listeners to return, but we also have some errors; log them
-		log.Info(err.Error())
+		log.Info(errs.Errors[0].Error())
+	} else if len(errs.Errors) > 1 {
+		log.Info(errs.Error())
```

Comment:
What are these change for?

Reply:
otherwise we print a long multi line error like

1 error occurred:
* some error
newline

instead of just the error

Reply:
Seems this only applies to one error, for more than one error, there are still many lines. It looks more clear with multi-lines.

This kind of usage is widespread across the whole project, if you feel the format is not good, isn't is more simple to file a fix in upstream hashicorp/go-multierror?

Reply:
May be we should have custom ErrorFormat func https://github.com/hashicorp/go-multierror/blob/master/multierror.go#L12 and slowly move other usages to this?

Reply:
good idea

Reply:
thanks, done